### PR TITLE
JavaScript Coding Standards

### PR DESCRIPTION
**Pull request by gailsteiger**
Added JavaScript coding standards and guidelines.

### CODE REVIEW

File: handbook.md [hunk 1]
```diff
@@ -125,6 +125,170 @@ We follow the [Ruby Style Guide](https://github.com/bbatsov/ruby-style-guide) an
...
+* ["space-unary-ops": ["error", { "words": true, "nonwords": false }]](http://eslint.org/docs/rules/space-unary-ops)
+* ["spaced-comment": ["error", "always"]](http://eslint.org/docs/rules/spaced-comment)
+* ["strict": "off"](http://eslint.org/docs/rules/strict)
+* ["use-isnan": "error"](http://eslint.org/docs/rules/use-isnan)
+* ["valid-jsdoc": "off"](http://eslint.org/docs/rules/valid-jsdoc)
+* ["valid-typeof": "error"](http://eslint.org/docs/rules/valid-typeof)
+* ["vars-on-top": "error"](http://eslint.org/docs/rules/vars-on-top)
+* ["wrap-iife": ["error", "any"]](http://eslint.org/docs/rules/wrap-iife)
+* ["wrap-regex": "off"](http://eslint.org/docs/rules/wrap-regex)
+* ["yoda": ["error", "never"]](http://eslint.org/docs/rules/yoda)
```
Comment (lzap) [inline handbook.md]: This list will quickly become obsolete as we will be modifying jslint configuration file. Can you provide a link to the configuration stored in git preferably instead?

File: handbook.md [hunk 2]
```diff
@@ -125,6 +125,170 @@ We follow the [Ruby Style Guide](https://github.com/bbatsov/ruby-style-guide) an
 * Use single character variable names or abbreviations. The keyboard erosion you might save by doing that is not worth it.
 * Squash exceptions
 
+### JavaScript
+We are supporting and encouraging the use of es2015. (For more information see [Learn ES2015](http://babeljs.io/docs/learn-es2015/).) For browser compatibility, we use Babel to transform the code to es5.  We use eslint to enforce linting rules which are detailed below.
```
Comment (dLobatog) [inline handbook.md]: Please add how to run eslint here, `npm run lint` should work

Comment (gailsteiger) [inline handbook.md]: done

File: handbook.md (hunk 1 above)
Comment (gailsteiger) [inline handbook.md]: done

File: handbook.md [hunk 3]
```diff
@@ -125,6 +125,25 @@ We follow the [Ruby Style Guide](https://github.com/bbatsov/ruby-style-guide) an
...
+* Add all new JavaScript files to the project in the 'webpack/assets/javascripts' folder or a sub-folder.
+* Use es2015 syntax and features.
+* Use exporting/importing values from/to modules to manage dependencies.
+
+#### Don't
+* Pollute the global namespace. If, for some reason, you must expose a function/value globally, use the ```window.tfm``` object as implemented in ```bundle.js```.
+
+#### Linting rules
+
+The latest lint rules for foreman can be found [here](https://github.com/gailsteiger/foreman/blob/ac536d922328299b4c33fd0d3dc0a10cf2ed7003/.eslintrc). See [ESLint Rules](http://eslint.org/docs/rules/) for more information regarding rule configuration.
```
Comment (dLobatog) [inline handbook.md]: That will have to point to that file in develop - let's wait until it's merged, I can change ito n merge

**Comment by dLobatog** [comment]
All good :+1: - will change the line that points to your branch currently to point to develop upon merge of https://github.com/theforeman/foreman/pull/3673 Thanks @gailsteiger !

**Comment by dLobatog** [comment]
Actually let's wait until https://github.com/theforeman/foreman/pull/3761/files is merged too, so you can say here that ESLint runs on every PR.

**Comment by dLobatog** [comment]
All ready now - please point to https://github.com/theforeman/foreman/blob/develop/.eslintrc in the doc instead of your fork, and this should be ready for merge. Thanks!

**Comment by gailsteiger** [comment]
@dLobatog done

**Comment by dLobatog** [comment]
Merged as a0679ce2cad5cf21f2d5a67a6a732d7e49f47887, thanks @gailsteiger !

**Pull request closed without merging**